// weaviate-client/dist/node/cjs/batch/index.js
'use strict';
var __importDefault =
  (this && this.__importDefault) ||
  function (mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, '__esModule', { value: true });
exports.ReferencesBatcher = exports.ObjectsBatchDeleter = exports.ObjectsBatcher = void 0;
const objectsBatcher_js_1 = __importDefault(require('./objectsBatcher.js'));
const objectsBatchDeleter_js_1 = __importDefault(require('./objectsBatchDeleter.js'));
const referencesBatcher_js_1 = __importDefault(require('./referencesBatcher.js'));
const referencePayloadBuilder_js_1 = __importDefault(require('./referencePayloadBuilder.js'));
const beaconPath_js_1 = require('../utils/beaconPath.js');
const batch = (client, dbVersionSupport) => {
  const beaconPath = new beaconPath_js_1.BeaconPath(dbVersionSupport);
  return {
    objectsBatcher: () => new objectsBatcher_js_1.default(client),
    objectsBatchDeleter: () => new objectsBatchDeleter_js_1.default(client),
    referencesBatcher: () => new referencesBatcher_js_1.default(client, beaconPath),
    referencePayloadBuilder: () => new referencePayloadBuilder_js_1.default(client),
  };
};
exports.default = batch;
var objectsBatcher_js_2 = require('./objectsBatcher.js');
Object.defineProperty(exports, 'ObjectsBatcher', {
  enumerable: true,
  get: function () {
    return __importDefault(objectsBatcher_js_2).default;
  },
});
var objectsBatchDeleter_js_2 = require('./objectsBatchDeleter.js');
Object.defineProperty(exports, 'ObjectsBatchDeleter', {
  enumerable: true,
  get: function () {
    return __importDefault(objectsBatchDeleter_js_2).default;
  },
});
var referencesBatcher_js_2 = require('./referencesBatcher.js');
Object.defineProperty(exports, 'ReferencesBatcher', {
  enumerable: true,
  get: function () {
    return __importDefault(referencesBatcher_js_2).default;
  },
});
